Update model-notifications (#6556)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/model-notifications.md

## Configure groups

> Define your groups in any .yml file in your [models directory](/reference/project-configs/model-paths). For example: 

Author: This is wrong and is the main thing I wanted to do in the PR

  - name: finance
    description: "Models related to the finance department"
    owner:
      # Email is required to receive model-level notifications, additional properties are also allowed.
      name: "Finance Team"
      email: finance@dbtlabs.com
      favorite_food: donuts

  - name: marketing
    description: "Models related to the marketing department"
    owner:
      name: "Marketing Team"
      email: marketing@dbtlabs.com

>       favorite_food: jaffles

Author: This is for clarity and is optional, but reduces the likelihood people misbelieve that Slack notifs are supported

Reviewer: good callout! i assumed slack was supported when reading

```

</File>

## Attach groups to models

> Attach groups to models as you would any other config, in either the `dbt_project.yml` or `<whatever>.yml` files. For example: 

Reviewer: ```suggestion
Attach groups to models as you would any other config, in either the `dbt_project.yml` or `whatever.yml` files. For example: 
```

> <File name='models/marts.yml'>

Author: The rest of this is purely additive and you can rework as you see fit

Reviewer: thanks for adding the files tag! 🙏

      group: marketing

```
</File>

> By assigning groups in the `dbt_project.yml` file, you can capture all models in a subdirectory at once. In the below example, all notifications related to staging models will go to the data engineering group, whereas models in `marts/sales` will be routed to the finance team.

Reviewer: ```suggestion
By assigning groups in the `dbt_project.yml` file, you can capture all models in a subdirectory at once. 

In this example, model notifications related to staging models go to the data engineering group, `marts/sales` models to the finance team, and `marts/campaigns` models to the marketing team.
```

<File name='dbt_project.yml'>

```yml
config-version: 2
name: "jaffle_shop"

[...]

models:
  jaffle_shop:
    staging:
      +group: data_engineering
    marts:
      sales:
        +group: finance
      campaigns:
        +group: marketing
    
```

</File>

> Attaching a group to a model also encompasses its tests, so you will also receive notifications for a model's test failures. 

Author: As is this
